### PR TITLE
Fix: Rare Scoreboard Errors

### PR DESCRIPTION
**Pull request by hannibal002**
## What
As seen in the image (created with https://github.com/hannibal002/SkyHanni/pull/2412), hypixel sometimes sends new scoreboard data packets delayed, giving us a invalid state of the current scoreboard for a brief moment in time. sometimes, this happens exactly at the moment when custom scoreboard pulls data. This causes weird/unlogical errors.

The fix is to wait 300ms (time can be varied maybe) after a change, and only then accept this change as valid. if a new change happens in between this time, the timer starts again.

<details>
<summary>Images</summary>

image

</details>

## Changelog Fixes
+ Fixed rare Custom Scoreboard errors. - hannibal2

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 1]
```diff
@@ -45,6 +48,13 @@ object CustomScoreboard {
     private var cache = emptyList<ScoreboardElementType>()
     private val guiName = "Custom Scoreboard"
 
+    // cached scoreboard data, only update after no change for 300ms
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]:
```suggestion
    // Cached scoreboard data, only update after no change for 300ms
```

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 2]
```diff
@@ -45,6 +48,13 @@ object CustomScoreboard {
...
+    var activeLines = emptyList<String>()
+
+    // most recent scoreboard state, not in use until cached
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]:
```suggestion
    // Most recent scoreboard state, not in use until cached
```

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 3]
```diff
@@ -90,8 +100,16 @@ object CustomScoreboard {
     fun onTick(event: LorenzTickEvent) {
         if (!isEnabled()) return
 
+        // we want to update the scoreboard as soon as we have new data, not 5 ticks delayed
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]:
```suggestion
        // We want to update the scoreboard as soon as we have new data, not 5 ticks delayed
```

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 4]
```diff
@@ -90,8 +100,16 @@ object CustomScoreboard {
...
+        var dirty = true
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]: I think this should be false

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt [hunk 5]
```diff
@@ -102,6 +120,13 @@ object CustomScoreboard {
         UnknownLinesHandler.handleUnknownLines()
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt:120]: The UnknownLinesHandler will probably still trigger on these broken lines

**Review by j10a1n15 — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt (hunk 5 above)
Comment (j10a1n15) [reply src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt:120]: image
this should be an easy fix though

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt (hunk 4 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt]: ufff. yes

File: src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt (hunk 5 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt:120]: good catch

**Review by j10a1n15 — APPROVED** [review]

**Pull request merged**